src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol

```diff
@@ -265,38 +260,28 @@ abstract contract SecurityCouncilMemberElectionGovernorCountingUpgradeable is
 
         // do i have an off-by-one in here?
 
-        uint256 startBlock = proposalSnapshot(proposalId);
         uint256 endBlock = proposalDeadline(proposalId);
-
-        uint256 duration = endBlock - startBlock;
+        uint256 startBlock = proposalSnapshot(proposalId);
 
         if (blockNumber <= startBlock || blockNumber > endBlock) {
             return 0;
         }
 
-        uint256 fullWeightDuration =
-            WAD * _fullWeightDurationNumerator / _durationDenominator * duration / WAD;
+        uint256 fullWeightVotingDeadline_ = fullWeightVotingDeadline(proposalId);
 
-        uint256 decreasingWeightStartBlock = startBlock + fullWeightDuration;
-
-        if (blockNumber <= decreasingWeightStartBlock) {
+        if (blockNumber <= fullWeightVotingDeadline_) {
             return votes;
         }
 
         // slope denominator
-        uint256 decreasingWeightDuration =
-            WAD * _decreasingWeightDurationNumerator / _durationDenominator * duration / WAD;
-
-        // slope numerator is -votes
-
-        uint256 decreaseAmount = WAD * votes / decreasingWeightDuration
-            * (blockNumber - decreasingWeightStartBlock) / WAD;
+        uint256 decreasingWeightDuration = endBlock - fullWeightVotingDeadline_;
```

Comment:
we should guarantee `fullWeightVotingDeadline_ < endBlock` by enforcing `_fullWeightDuration <= votingDuration` on init

Reply:
good point, done here cf8d157308c26fef11b95e2046a45f2928ded4b8

Reply:
if you think it's better, i could flip these in the governor init and do the require in the counting module init

```
__SecurityCouncilMemberElectionGovernorCounting_init({
    maxNominees: _maxNominees,
    initialFullWeightDuration: _fullWeightDuration
});
__GovernorSettings_init(0, _votingPeriod, 0);
```